nexus/mgs-updates/src/common_sp_update.rs

```diff
@@ -319,6 +321,18 @@ pub enum PrecheckError {
     WrongInactiveVersion { expected: ExpectedVersion, found: FoundVersion },
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum PostUpdateError {
+    #[error("communicating with MGS")]
+    GatewayClientError(#[from] GatewayClientError),
+
+    #[error("communicating with RoT: {message:?}")]
+    RotCommunicationFailed { message: String },
+
+    #[error("invalid RoT bootloader image: {error:?}")]
+    RotBootloaderImageError { error: RotImageError },
```

Comment:
I'm curious for @jgallagher's take on this but it would seem nice to me if the generic parts of this package (this file, the driver, and `apply_update`) didn't know so much about specific devices.  This would preclude this type from including more specific typed errors like `RotImageError`, but I believe the only thing consumers of this error type care about is that the error is fatal to the update attempt.

So I'd consider renaming `RotCommunicationFailed` to `TransientError` and `RotBootloaderImageError` to `FatalError`.  Both would just contain `message: String`.